### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by maryanchuk04**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk 
- [ ] @sand0 

### Second Level Review

- [ ] @2bbodia 

## Summary of issue

Finish work with general information on User Profile

## Summary of change

Add Services & Controllers for working with General info and connect with FrontEnd

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js [hunk 1]
```diff
@@ -52,8 +53,14 @@ const EditBirthday = ({ handleSubmit, pristine, reset, submitting, error }) => {
...
-            <div>
-                <Button
+            <div className='editFieldButtons'>
+                <IconButton className="text-success" size="small" type="submit" disabled={pristine || submitting} >
+                        <i className="fas fa-check" />
+                </IconButton>
+                <IconButton className="text-danger" size="small" onClick={reset && onClose}>
+                    <i className="fas fa-times" />
+                </IconButton>
+                {/* <Button
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js]: Remove commented code.

File: EventsExpress/ClientApp/src/components/profile/editProfile/editFieldsStyles.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.field {
+    position: relative;
+}
+
+.content {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editFieldsStyles.css]: Remove empty class

File: EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js [hunk 3]
```diff
@@ -26,10 +27,16 @@ let EditGender = props => {
                 }
             </div>
             
-            <div>
-                <Button type="submit" color="primary" disabled={pristine || submitting}>
+            <div className='editFieldButtons'>
+                {/* <Button type="submit" color="primary" disabled={pristine || submitting}>
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js]: Remove commented code.

File: EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js [hunk 4]
```diff
@@ -33,7 +34,7 @@ const validate = values => {
     return errors;
 };
 
-const EditBirthday = ({ handleSubmit, pristine, reset, submitting, error }) => {
+const EditBirthday = ({ handleSubmit, pristine, reset, submitting, error, onClose }) => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js]: All of the "Edit..." components share similar structure. Let's extract similar functionality into a reusable component.

File: EventsExpress/ClientApp/src/components/profile/profile-sections/info-field.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        }
+    }
+
+
+    let Element = editContainer;
+    return(
+        <div className={classes.fieldStyle} onClick={openField}>
+            {
+                showEdit ? <>
+                    {!isOpen &&
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/profile-sections/info-field.js]: Combine the two condition into single render and extract common functionality into a method.

File: EventsExpress/ClientApp/src/components/profile/profile.js [hunk 6]
```diff
@@ -41,10 +80,24 @@ const Profile = (props) => {
     const handleChange = panel => (event, isExpanded) => {
         setExpanded(isExpanded ? panel : false);
     };
-
+    console.log(props);
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/profile.js]: Remove console.log.

File: EventsExpress/ClientApp/src/components/profile/profile.js [hunk 7]
```diff
@@ -41,10 +80,24 @@ const Profile = (props) => {
...
+                    <InterestsSection />
+                </div>
+                <div className={classes.moreInfoBlock}>
+                    <MoreInfoSection />
+                </div>
+                <div className={classes.generalInfoBlock}>
+                    <GeneralInfoSection />
+                </div>
+            </div>
+            {/* <ExpansionPanel expanded={expanded === 'panel0'} onChange={handleChange('panel0')}>
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/profile.js]: Remove commented code.

File: EventsExpress/ViewModels/EditFirstNameViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace EventsExpress.ViewModels;
+
+public class EditFirstNameViewModel
```
Comment (iuriikhrystiuk) [inline EventsExpress/ViewModels/EditFirstNameViewModel.cs:3]: Consider creating validators for the new ViewModels.

File: EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js [hunk 9]
```diff
@@ -1,80 +1,34 @@
...
+import IconButton from "@material-ui/core/IconButton";
 import moment from 'moment';
 import ErrorMessages from '../../shared/errorMessage';
 import { renderDatePicker, parseEuDate } from '../../helpers/form-helpers';
 import { fieldIsRequired } from '../../helpers/validators/required-fields-validator';
+import './editFieldsStyles.css'
+import EditBaseProfile from './editProfilePropertyWrapper';
 
-const today = () => moment().startOf('day');
+// //EditProfilePropertyWrapper
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js]: Remove commented code. Are you using this component at all?

File: EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js [hunk 10]
```diff
@@ -3,9 +3,10 @@ import { Field, reduxForm } from "redux-form";
 import Button from "@material-ui/core/Button";
 import { renderSelectField } from '../../helpers/form-helpers'
 import ErrorMessages from '../../shared/errorMessage';
+import IconButton from "@material-ui/core/IconButton";
 
 let EditGender = props => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js]: Is it nor possible to refactor edit gender?

Comment (maryanchuk04) [reply EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js]: This component is not used, remove it?

File: EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js (hunk 9 above)
Comment (maryanchuk04) [reply EventsExpress/ClientApp/src/components/profile/editProfile/editBirthday.js]: This too.
This logic is present in the container

File: EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js (hunk 10 above)
Comment (maryanchuk04) [reply EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js]: Maybe I can delete all the files Edit...?

Comment (iuriikhrystiuk) [reply EventsExpress/ClientApp/src/components/profile/editProfile/editGender.js]: Yes, remove unused code please.

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**